dynamodb-kv/capacity.c

```diff
@@ -130,9 +131,15 @@ callback_readmetadata(void * cookie, struct http_response * res)
 
 		/* Get ReadCapacityUnits and WriteCapacityUnits. */
 		capstr = json_find(buf, end, "ReadCapacityUnits");
-		capr = strtol((const char *)capstr, NULL, 10);
+		if (PARSENUM_EX(&capr, (const char *)capstr, 10, 1)) {
```

Comment:
I'm not sure how this could possibly work.  `capr` and `capw` are signed integers, right?

Reply:
You declared them as
```
    long capr, caprw;
```
at the top of `callback_readmetadata()`.

Reply:
Right, and `long` is a signed type, so we can't use PARSENUM on it without explicit bounds?

Reply:
oh crap.

... right, so it's interpreting `10, 1` as the bounds, not as the base and trailing.  Ouch.  I'll need to do something about that.